src/panel/direction/DirectionPanel.jsx

```diff
@@ -114,22 +114,25 @@ export default class DirectionPanel extends React.Component {
         destination,
         vehicle,
       );
-      if (directionResponse && directionResponse.routes && directionResponse.routes.length > 0) {
-        const routes = directionResponse.routes.map((route, i) => ({
+      if (directionResponse && directionResponse.error === 0) {
+        // Valid, non-empty response
+        const routes = directionResponse.data.routes.map((route, i) => ({
           ...route,
           isActive: i === 0,
           id: i,
         }));
-        this.setState({ isLoading: false, error: false, routes });
+        this.setState({ isLoading: false, error: 0, routes });
         window.execOnMapLoaded(() => {
           fire('set_route', { routes, vehicle, origin, destination });
         });
       } else {
-        this.setState({ isLoading: true, error: true });
+        // Error or empty response
+        this.setState({ isLoading: false, error: directionResponse.error });
         fire('clean_route');
       }
     } else {
-      this.setState({ isDirty: false, routes: [] });
+      // When both fields are not filled yet or not filled anymore
+      this.setState({ isLoading: false, isDirty: false, error: 0, routes: [] });
```

Comment:
we should also set `error: 0` on line 111, to avoid showing an obsolete error when a new request is triggered.